Move YSENTRY into the base and rename it (#1066)
riscv/riscv-cheri: src/cheri/rvy64-encoding.adoc

The capabilities of this chapter define a 1-bit field for <<sec_cap_type>> values;
this field directly encodes the values `0` and `1`:

* The encoded value `0` indicates an _unsealed_ type.

> * The encoded value `1` indicates an _sealed_ type which can unsealed by <<JALR_CHERI>>.

Reviewer: ```suggestion
* The encoded value `1` indicates a <<sentry_cap>> which can unsealed by <<JALR_CHERI>>.
```
